### ISSUE TITLE
Error during `amplify init` process, unable to init a project

### ISSUE THREAD

**Issue by trusktr** [aws-amplify/amplify-cli#656]
**Describe the bug**

Unable to init a project.

**To Reproduce**
1. Go into an existing TypeScript project
2. run `npm install @aws-amplify/cli`
3. run `./node_modules/.bin/amplify init`
4. answer questions.
5. When it gets to question `Choose the type of that you're building (Use arrow keys)`, it shows no options. Pressing arrow keys does nothing (because there's no options).
6. Pressing enter causes an error as follows:

```
❯ ./node_modules/.bin/amplify init
Note: It is recommended to run this command from the root of your app directory
? Enter a name for the project test
? Choose your default editor: Atom Editor
? Choose the type of app that you're building

/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/gluegun/build/index.js:13
    throw up;
    ^

TypeError: Cannot read property 'value' of undefined
    at ListPrompt.getCurrentValue (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/@aws-amplify/cli/node_modules/inquirer/lib/prompts/list.js:121:53)
    at MapSubscriber._next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/operators/map.js:49:35)
    at MapSubscriber.Subscriber.next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/Subscriber.js:67:18)
    at TakeSubscriber._next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/operators/take.js:54:30)
    at TakeSubscriber.Subscriber.next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/Subscriber.js:67:18)
    at Interface.handler (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/observable/fromEvent.js:22:28)
    at Interface.emit (events.js:187:15)
    at Interface._onLine (readline.js:290:10)
    at Interface._line (readline.js:638:8)
    at Interface._ttyWrite (readline.js:919:14)
```

Here's what it looks like right when I get to the third prompt, before I press any arrow keys:

```
Signafy+mapper-annotated-scene git:feature/annotation-labels ❯ ./node_modules/.bin/amplify init
Note: It is recommended to run this command from the root of your app directory
? Enter a name for the project test
? Choose your default editor: Atom Editor
? Choose the type of app that you're building (Use arrow keys)
```

and there's no options visible under that third prompt.

**Expected behavior**

To get through init process

**Screenshots**

<img width="799" alt="screen shot 2019-01-03 at 1 46 24 pm" src="https://user-images.githubusercontent.com/297678/50663275-f9de9580-0f5d-11e9-94a2-1e54ef4388e2.png">

**Desktop (please complete the following information):**

 - OS: macOS Mojave
 - Version:

```
❯ cat node_modules/@aws-amplify/cli/package.json | grep version
  "version": "0.1.38"
```

**Smartphone (please complete the following information):**

N/A

**Additional context**

Let me know if you need any

**Comment by kaustavghosh06** [comment]
@trusktr I wasn't able to unfortunately reproduce the issue on my Mac when I locally install the CLI
![screen shot 2019-01-03 at 1 59 15 pm](https://user-images.githubusercontent.com/7247658/50663839-dae10300-0f5f-11e9-9089-5c41a28d1959.png)

To unblock yourself for now, could you install the CLI globally using `npm install -g @aws-amplify/cli`? Trying to investigate further to rootcause this behavior.

**Comment by trusktr** [comment]
@kaustavghosh06 Sure, let me try that.

I also tried making a new folder inside my project, then tried it again, but also got the same error (though with one less prompt):

```
❯ ../node_modules/.bin/amplify init # note the `../` this time.
Note: It is recommended to run this command from the root of your app directory
? Choose your default editor: Atom Editor
? Choose the type of app that you're building

/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/gluegun/build/index.js:13
    throw up;
    ^

TypeError: Cannot read property 'value' of undefined
    at ListPrompt.getCurrentValue (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/@aws-amplify/cli/node_modules/inquirer/lib/prompts/list.js:121:53)
    at MapSubscriber._next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/operators/map.js:49:35)
    at MapSubscriber.Subscriber.next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/Subscriber.js:67:18)
    at TakeSubscriber._next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/operators/take.js:54:30)
    at TakeSubscriber.Subscriber.next (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/Subscriber.js:67:18)
    at Interface.handler (/Users/trusktr/src/Signafy+mapper-annotated-scene/node_modules/rxjs/internal/observable/fromEvent.js:22:28)
    at Interface.emit (events.js:187:15)
    at Interface._onLine (readline.js:290:10)
    at Interface._line (readline.js:638:8)
    at Interface._ttyWrite (readline.js:919:14)
```

**Comment by trusktr** [comment]
Alright, I've confirmed, if I install it globally, it works fine.

This is a problem because I don't install tools globally; that makes it difficult to manage version across projects, and I don't want to update all projects when a newer projects works with a newer CLI and the others don't.

So what I do is install things locally inside each project, and run the script from `node_modules/.bin`, and then add shortcuts to package.json `scripts`.

**Comment by kaustavghosh06** [comment]
@trusktr It should ideally work locally as well as I showed in my screenshot above (it works for me on my machine when I install it locally inside a test project).
To investigate further, could you send a screenshot of your node_modules/ when you install the CLI locally? The problem is that the CLI is not able to find the dependent plugins (frontend plugins) for some reason because of which it seems to hang in your case.

**Comment by kaustavghosh06** [comment]
@trusktr I tried installing it locally on various machines (Linux based) and it works fine with the latest version of the CLI. Let me know if you're still facing this issue when installing the CLI locally. I would love to see your local/node_modules structures if you're still seeing the issue.

**Comment by kaustavghosh06** [comment]
We published a fix for this in our version 1.1.8. Please take a look at it and re-open this issue if you still are facing problems.

**Comment by melwinpais** [comment]
Team,

I have a related issue: **./node_modules/.bin/amplify init** doesnt behave like **amplify init**
I need to execute it off project folder so that I can use it in CI/CD pipeline.

Let me know if there is a fix/workaround.

```
instance:my-amply-app paism$ ./node_modules/.bin/amplify init --amplify '{"projectName":"connect-console","envName":"yarra","defaultEditor":"code"}' --frontend '{"frontend":"javascript","framework":"react","config":{"SourceDir":"src","DistributionDir":"build","BuildCommand":"npm run-script build","StartCommand":"npm run-script start"}}' --providers '{"awscloudformation":{"configLevel":"project","useProfile":true,"profileName":"test","region":"ap-southeast-2"}}' --yes
Scanning for plugins...
Plugin scan successful
Note: It is recommended to run this command from the root of your app directory
Using default provider  undefined
init failed
TypeError [ERR_INVALID_ARG_TYPE]: The 'request' argument must be string. Received type undefined
    at /Users/paism/ws/tab/my-amply-app/node_modules/@aws-amplify/cli/lib/lib/init-steps/s2-initProviders.js:55:46
    at Array.forEach (<anonymous>)
    at /Users/paism/ws/tab/my-amply-app/node_modules/@aws-amplify/cli/lib/lib/init-steps/s2-initProviders.js:54:31
    at step (/Users/paism/ws/tab/my-amply-app/node_modules/@aws-amplify/cli/lib/lib/init-steps/s2-initProviders.js:33:23)
    at Object.next (/Users/paism/ws/tab/my-amply-app/node_modules/@aws-amplify/cli/lib/lib/init-steps/s2-initProviders.js:14:53)
    at fulfilled (/Users/paism/ws/tab/my-amply-app/node_modules/@aws-amplify/cli/lib/lib/init-steps/s2-initProviders.js:5:58)
instance:my-amply-app paism$ amplify init --amplify '{"projectName":"connect-console","envName":"yarra","defaultEditor":"code"}' --frontend '{"frontend":"javascript","framework":"react","config":{"SourceDir":"src","DistributionDir":"build","BuildCommand":"npm run-script build","StartCommand":"npm run-script start"}}' --providers '{"awscloudformation":{"configLevel":"project","useProfile":true,"profileName":"test","region":"ap-southeast-2"}}' --yes
Scanning for plugins...
Plugin scan successful
Note: It is recommended to run this command from the root of your app directory

For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-multiple-profiles.html
... WORKS FINE ...

```
